src/components/nodes/signalEventDefinition/index.js

```diff
@@ -22,38 +22,13 @@ export default {
     }
     
     let signal = definitions.rootElements.find(element => element.id === value.signalRef);
-    if (!signal) {
+    if (!signal && value.signalRef) {
       signal = moddle.create('bpmn:Signal', {
         id: value.signalRef,
         name: value.signalRef,
       });
       definitions.rootElements.push(signal);
```

Comment:
Nice suggestion I will add the validation to the validation property.
In other hand, looks like the regular expression used in `modeler/src/components/inspectors/idConfigSettings.js` is wrong. Reviewing the W3C specifications the right expression should be:
```
[\i-[:]][\c-[:]]*
```
https://www.w3.org/TR/xmlschema11-2/#IDREF
for javascript implementation should be:
```
[_A-Za-z][-._A-Za-z0-9]*
```

Reply:
@caleeli I thought the ID was suppose to be a QName: https://www.w3.org/TR/xmlschema11-2/#QName. Which seems to have a slightly different pattern:
```regex
[a-zA-Z]{1,8}(-[a-zA-Z0-9]{1,8})*
```

Reply:
@chipit24 Reviewing the BPMN xsd schemas (from Semantic.xsd) I could find:
All the IDs must be `xsd:ID` since all the BPMN elements inherit its structure from BaseElement:
```
	<xsd:element name="baseElement" type="tBaseElement"/>
	<xsd:complexType name="tBaseElement" abstract="true">
		<xsd:sequence>
			<xsd:element ref="documentation" minOccurs="0" maxOccurs="unbounded"/>
			<xsd:element ref="extensionElements" minOccurs="0" maxOccurs="1" /> 
		</xsd:sequence>
		<xsd:attribute name="id" type="xsd:ID" use="optional"/>
		<xsd:anyAttribute namespace="##other" processContents="lax"/>
	</xsd:complexType>
```
And most of the "Refs" use `xsd:QName`
```
	<xsd:element name="signalEventDefinition" type="tSignalEventDefinition" substitutionGroup="eventDefinition"/>
	<xsd:complexType name="tSignalEventDefinition">
		<xsd:complexContent>
			<xsd:extension base="tEventDefinition">
				<xsd:attribute name="signalRef" type="xsd:QName"/>
			</xsd:extension>
		</xsd:complexContent>
	</xsd:complexType>
```
Well the main difference between xsd:ID and xsd:QName  is that QName could have a domain prefix in its value like `pm:`, for example: `pm:script_5`